Feat: add some documents about the user and RBAC (#597)
oam-dev/kubevela.io: docs/how-to/dashboard/user/project.md

---
title: Project management
---

Project provides a logical separation of applications、environments and delivery targets, this is helpful when VelaUX is used by multiple teams. Project can provide the following features:

> * restrict where applications may be deployed to (target clusters and namespaces).
> * restrict what integration configs can be used (trusted Helm repositories, docker registries or cloud providers).
> * restrict the different permission of each user, and provide a multiple-level permission control (RBAC).

Reviewer: ```suggestion
* restrict where applications can be deployed to (target clusters and namespaces).
* restrict what kinds of integrated configration can be used (trusted Helm repositories, docker registries or cloud providers).
* restrict the different permission of each user, and provide a multiple-level permission control (RBAC).
```
```suggestion
* restrict where applications may be deployed to (target clusters and namespaces).
* restrict what integration configs can be used (trusted Helm repositories, docker registries or cloud providers).
* restrict the different permission of each user, and provide a multiple-level permission control (RBAC).
```

> * restrict where applications may be deployed to (target clusters and namespaces).
> * restrict what integration configs can be used (trusted Helm repositories, docker registries or cloud providers).
> * restrict the different permission of each user, and provide a multiple-level permission control (RBAC).

Reviewer: ```suggestion
* restrict where applications can be deployed to (target clusters and namespaces).
* restrict what kinds of integrated configration can be used (trusted Helm repositories, docker registries or cloud providers).
* restrict the different permission of each user, and provide a multiple-level permission control (RBAC).
```
```suggestion
* restrict where applications may be deployed to (target clusters and namespaces).
* restrict what integration configs can be used (trusted Helm repositories, docker registries or cloud providers).
* restrict the different permission of each user, and provide a multiple-level permission control (RBAC).
```

---
title: Project management
---

> Projects provide a logical grouping of applications、environments and targets, which is useful when VelaUX is used by multiple teams. Projects provide the following features:

Reviewer: ```suggestion
Project provides a logical separation of applications、environments and delivery targets, this is helpful when VelaUX is used by different teams. Project can provide the following features:
```

> * restrict where apps may be deployed to (target clusters and namespaces), we plan to support setting resource limits for the target.

Reviewer: ```suggestion
* applications can be deployed to the target clusters and namespaces belong to the same project.
```

没实现的就先不写上去了。

Author: restrict where applications may be deployed to (target clusters and namespaces).

这个表达我认为更准确。

> * restrict what Integrations can be use(trusted Helm repositories, docker registry or cloud provider)

Reviewer: ```suggestion
* restrict which kinds of integration can be use, you can specify config within a project that contains trusted Helm repositories, docker registries or cloud providers.
```

注意单复数，和末尾句号。

Author: restrict what integration configs can be used (trusted Helm repositories, docker registries or cloud providers).

> * defining project roles to provide application RBAC.

Reviewer: ```suggestion
* project level RBAC, the project administrator can control the application level authorization.
```

Author: restrict the different permission of each user, and provide a multiple-level permission control (RBAC).

> * restrict what kinds of objects or component definitions may or may not be deployed(TODO)

Reviewer: ```suggestion
```

没实现的没必要写到文档，写到Roadmap去。

## The Default Project

> Once installed VelaUX has one default project with name `Default`, and owner by admin user. At the same time, the default target and environment are also generated by default and bound in the project.

Reviewer: ```suggestion
Once installed VelaUX has one default project with name `Default` owned by the `admin` user. At the same time, the default target and environment are also generated automatically and bound to the project.
```

## Creating Projects

> Users with project management rights can go to `Platform/Projects` page，in this page you could create and delete project.

Reviewer: ```suggestion
Users with project management permissions can go to `Platform/Projects` page for settings，in this page you could create and delete project.
```

![project-list](https://static.kubevela.net/images/1.3/project-list.jpg)

> Click the `New Project` button, you can open the drawer page for creating a project. Each project should set a name and owner, the owner user can auto grant the project admin role after the project is created.

Reviewer: ```suggestion
Click the `New Project` button for creating a project. Each project should set name and owner, the owner user will be granted the project admin role automatically after the project created.
```

## Updating Projects

> Project owner, alias, and description information support updating. Click the project name and go to the project detail page, you can manage the members and roles in this project.

Reviewer: ```suggestion
Project owner, alias, and description fields can be updated. Click the project name and go to the project detail page, you can manage the members and roles in this project.
```

## Deleting Projects

> If there are applications and environments in a project, it can not be deleted. you must delete applications, environments, and targets in the project that you want to delete.

Reviewer: ```suggestion
If there are applications and environments living in one project, then deleting the project is not allowed. You must clean up the applications, environments, and targets in the project before you can delete.
```
